### PR TITLE
2023.1 Support

### PR DESCRIPTION
**Pull request by steviegt6**
This uses the same logic used in UMT, which I've tested in both this fork for contributing as well as my own independent fork. Tested against the latest version of HoloCure.

### CODE REVIEW

File: DogScepterLib/Core/Models/GMTextureGroupInfo.cs
Comment (colinator27) [inline DogScepterLib/Core/Models/GMTextureGroupInfo.cs:1]: There definitely seems to be some errors here with serializing. I think it's supposed to just not write anything to do with Spine (including the initial pointer up above).

File: DogScepterLib/Core/Chunks/GMChunkTGIN.cs [hunk 1]
```diff
@@ -57,6 +57,30 @@ private void DoFormatCheck(GMDataReader reader)
...
+            {
+                int returnTo = reader.Offset;
+                reader.Offset += 4; // Skip count.
+
+                uint firstPtr = reader.ReadUInt32();
+
+                // Navigate to the fourth list pointer, which is different
+                // depending on whether this is 2023.1+ or not (either "FontIDs"
+                // or "SpineSpriteIDs").
+                reader.Offset = (int)(firstPtr + 16 + (sizeof(uint) * 3));
```
Comment (colinator27) [inline DogScepterLib/Core/Chunks/GMChunkTGIN.cs:71]: Hmm... honestly unsure if sizeof() is a good idea to be using here, as a constant 4 bytes is applied everywhere else in the project (including in this code). Can probably just change that to 4, but unsure.

**Review by colinator27 — CHANGES_REQUESTED** [review]
Overall looks good, will probably just need some small tweaks to merge it in properly. (Most importantly making sure that serialization works properly.)

File: DogScepterLib/Core/Models/GMTextureGroupInfo.cs
Comment (steviegt6) [reply DogScepterLib/Core/Models/GMTextureGroupInfo.cs:1]: I'll have a look tomorrow, thanks.

File: DogScepterLib/Core/Chunks/GMChunkTGIN.cs (hunk 1 above)
Comment (steviegt6) [reply DogScepterLib/Core/Chunks/GMChunkTGIN.cs:71]: Doesn't really matter, suppose it's just a difference in codestyle -- it gets compiled directly to four bytes anyway, treated as a constant value.

**Review by colinator27 — APPROVED** [review]
Seems good now! Thanks again.

**Pull request merged**